orc/src/main/java/org/apache/iceberg/orc/OrcFileAppender.java

```diff
@@ -95,24 +97,16 @@ public Metrics metrics() {
       // we don't currently have columnSizes or distinct counts.
       Map<Integer, Long> valueCounts = new HashMap<>();
       Map<Integer, Long> nullCounts = new HashMap<>();
-      Integer[] icebergIds = new Integer[orcSchema.getMaximumId() + 1];
-      for (TypeDescription type : columnIds.keySet()) {
-        icebergIds[type.getId()] = columnIds.get(type);
-      }
-      for (int c = 1; c < stats.length; ++c) {
-        if (icebergIds[c] != null) {
-          valueCounts.put(icebergIds[c], stats[c].getNumberOfValues());
-        }
-      }
-      for (TypeDescription child : orcSchema.getChildren()) {
-        int childId = child.getId();
-        if (icebergIds[childId] != null) {
-          nullCounts.put(icebergIds[childId], rows - stats[childId].getNumberOfValues());
-        }
+      Map<ColumnMap.IcebergColumn, TypeDescription> icebertToOrc = columnIds.inverse();
+      for (Types.NestedField field : schema.columns()) {
```

Comment:
It seems that here we can iterate using the `icebergToOrc` instead of iterating using Iceberg schema and then constructing keys for `icebergToOrc` map, no?

Reply:
I've avoided constructing keys by using `Integer` for the column mapping. I think this is more straightforward.